modules/setting/setting.go

```diff
@@ -371,14 +371,17 @@ var (
 		AccessTokenExpirationTime  int64
 		RefreshTokenExpirationTime int64
 		InvalidateRefreshTokens    bool
-		JWTSecretBytes             []byte `ini:"-"`
+		JWTSigningAlgorithm        string `ini:"JWT_SIGNING_ALGORITHM"`
 		JWTSecretBase64            string `ini:"JWT_SECRET"`
+		JWTSigningPrivateKeyFile   string `ini:"JWT_SIGNING_PRIVATE_KEY_FILE"`
 		MaxTokenLength             int
 	}{
 		Enable:                     true,
 		AccessTokenExpirationTime:  3600,
 		RefreshTokenExpirationTime: 730,
 		InvalidateRefreshTokens:    false,
+		JWTSigningAlgorithm:        "RS256",
```

Comment:
this change default alg to async - not sure if it is breaking ...

Reply:
```suggestion
		JWTSigningAlgorithm:        "HS256",
```

Reply:
as per KN4CK3R ...
`https://accounts.google.com/.well-known/openid-configuration` and other big tech have **RS256** - so go for it

Reply:
For clarification:
RS256 is defined as "create the signature with SHA**256** and use the provided RSA key as secret. The RSA key is allowed to have any size and **not** just 256 bit. If Gitea generates a key the default size is 4096 bit.